plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectPushdown.java

```diff
@@ -106,9 +106,10 @@ private static boolean isInputFormatSupported(Properties schema)
     public static boolean isCompressionCodecSupported(InputFormat<?, ?> inputFormat, Path path)
     {
         if (inputFormat instanceof TextInputFormat) {
+            // S3 Select supports the following formats: uncompressed, GZIP and BZIP2.
             return getCompressionCodec((TextInputFormat) inputFormat, path)
                     .map(codec -> (codec instanceof GzipCodec) || (codec instanceof BZip2Codec))
-                    .orElse(false); // TODO (https://github.com/trinodb/trino/issues/2475) fix S3 Select when file not compressed
+                    .orElse(true);
```

Comment:
question: What if a file is compressed, but with a codec that is not supported? Maybe something like 

```
Optional<Codec> codec = getCompressionCodec((TextInputFormat) inputFormat, path);
if(codec.isEmpty()){
  // assume uncompressed
  return true;
}
```

Also I wonder how safe is to assume that a file is uncompressed if a codec is not found for a given extension? (I guess it is as expected as I see [similar assumptions](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectLineRecordReader.java#L121) being made in other parts of the code, but want to clarify).

Reply:
Hi Andrii, if a file is compressed with a codec that isn't supported, the codec would be different and this would return false:
```
codec -> (codec instanceof GzipCodec) || (codec instanceof BZip2Codec)
```
The default `orElse(true)` is only effective when the codec is null. So the method returns true only when codec is null (uncompressed), Gzip or Bzip2. We also have unit tests for the `isCompressionCodecSupported` method with different codec inputs.

Good point about the null codec assumption, though. I think it's reasonable- if there is no codec defined when a Hive table is created, it doesn't depend on codecs and is expected to have uncompressed files. This method internally uses Hadoop's [CompressionCodecFactory](https://hadoop.apache.org/docs/r2.4.1/api/org/apache/hadoop/io/compress/CompressionCodecFactory.html), which I think is the standard.

Thank you so much for looking into this PR!